src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -346,7 +346,7 @@ private void SynthesizeClosureEnvironments(ArrayBuilder<ClosureDebugInfo> closur
                         AddSynthesizedMethod(
                             frame.Constructor,
                             FlowAnalysisPass.AppendImplicitReturn(
-                                MethodCompiler.BindMethodBody(frame.Constructor, CompilationState, null),
+                                MethodCompiler.BindMethodBody(frame.Constructor, CompilationState, Diagnostics),
```

Comment:
I'm not sure what this change does in the context of this PR?

Reply:
The change to binding to call GenerateSynthesizedBodyStatements on all synthesized instance constructors is causing some synthesized methods to pass thorugh locations that they weren't before, and the null diagnostics were being yanked on and it was crashing.